src/parquet/column/writer.h

```diff
@@ -189,6 +190,22 @@ inline void TypedColumnWriter<DType>::WriteBatch(int64_t num_values,
   if (current_encoder_->EstimatedDataEncodedSize() >= properties_->data_pagesize()) {
```

Comment:
The values are already encoded  `WriteValues()` at this point. We could be going way above the limit `properties_->data_pagesize()` if `num_values` is high.